src/System.IO.Pipelines.Testing/BufferUtilities.cs

```diff
@@ -35,7 +35,8 @@ public static ReadableBuffer CreateBuffer(params byte[][] inputs)
 
                 // Create a segment that has offset relative to the OwnedMemory and OwnedMemory itself has offset relative to array
                 var ownedBuffer = new OwnedArray<byte>(chars);
-                var current = new BufferSegment(ownedBuffer, length, length * 2);
```

Comment:
I kinda like keeping to old ctors, but I could be convinced otherwise.

Reply:
Old constructors + Set/Reset methods?

Reply:
Yeah, but only do it if *you* think it's better too.